servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -366,6 +366,8 @@ void sourceTerminated(@Nullable Throwable cause) {
             }
             this.failureCause = cause;
             setFlag(SOURCE_TERMINATED);
+            // Mark the subscription as CANCELLED to prevent propagating cancel from channelClosed
+            subscriptionUpdater.getAndSet(WriteStreamSubscriber.this, CANCELLED);
```

Comment:
can this just be a regular volatile write since we don't need the return value? if not plz add a comment.

Reply:
sure, we don't need the old value. how do you think, should we prevent from changing `null` to `CANCELLED`?

Reply:
probably not, because we can not reach this state without emitting any signals from the publisher => we always have a subscription at this point

Reply:
Or just a `set()` if the intention is to avoid mixing use of field and updater.

Reply:
`set` does additional work to validate access and check the value. We are not consistent in other places anyway.